geonode/base/models.py

```diff
@@ -354,54 +354,85 @@ def dump_bulk_tree(cls, user, parent=None, keep_ids=True, type=None):
                 user,
                 'base.view_resourcebase'
             )
+
         resources = resources.filter(
-            polymorphic_ctype__model__in=ctype_filter,
+            polymorphic_ctype__model__in=resource_types,
         )
 
-        # Resources should be clean & public.
-        # Refer: set_default_permissions() and clear_dirty_state()
+        if resource_name is not None:
+            resources = resources.filter(title=resource_name)
+
         resources = get_visible_resources(
             resources,
             user,
             admin_approval_required=settings.ADMIN_MODERATE_UPLOADS,
             unpublished_not_visible=settings.RESOURCE_PUBLISHING,
             private_groups_not_visibile=settings.GROUP_PRIVATE_RESOURCES)
 
-        ret = []
-
-        for h_keyword in qset.order_by('name'):
-            serialized_hkw = serializers.serialize('python', [h_keyword])[0]
-
-            # django's serializer stores the attributes in 'fields'
-            fields = serialized_hkw['fields']
+        tree = {}
 
+        for hkw in qset.order_by('name'):
+            slug = hkw.slug
             tags_count = 0
-            tags_count = TaggedContentItem.objects.filter(
-                content_object__in=resources,
-                tag=HierarchicalKeyword.objects.get(slug=fields['slug'])).count()
 
-            if tags_count > 0:
-                fields['text'] = fields.pop('name')
-                fields['href'] = fields.pop('slug')
-                fields['tags'] = [tags_count]
-                del fields['path']
-                del fields['numchild']
-                del fields['depth']
+            tags = TaggedContentItem.objects.filter(
+                    content_object__in=resources,
+                    tag=hkw
+                    )
 
-                if 'id' in fields:
-                    # this happens immediately after a load_bulk
-                    del fields['id']
+            if tags.exists():
+                tags_count = tags.count()
```

Comment:
You can use directly `tags.count()` since you are using a `filter` for the queryset. This means returning an empty queryset `<MultilingualPolymorphicQuerySet []>` so the count will always be 0.

Reply:
You are right @mattiagiupponi. I wrongly intuited that on null query it will raise a count() on none. Thanks